File: leetcode/003-Longest-Substring-Without-Repeating-Characters/go/Solution.go
func lengthOfLongestSubstring(s string) int {
	res, leftmost := 0, 0

	var chars [255]int
	for i, _ := range chars {
		chars[i] = -1
	}

	for i, c := range s {
	    leftmost = max(leftmost, chars[int(c)]+1)
		res = max(res, i - leftmost + 1)
		chars[int(c)] = i
	}

	return res
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: leetcode/003-Longest-Substring-Without-Repeating-Characters/scala/Solution.scala
object Solution {
  def lengthOfLongestSubstring(s: String): Int = {
    var left_most_idx, tmp_max_len, max_len: Int = 0
    var chars = Array.fill(255)(-1)

    for ((c, i) <- s.view.zipWithIndex) {
      if (chars(c.toInt) + 1 > left_most_idx) {
        left_most_idx = chars(c.toInt) + 1
      }
      tmp_max_len = i - left_most_idx + 1
      if (tmp_max_len > max_len) max_len = tmp_max_len
      chars(c.toInt) = i
    }
    return max_len
  }
}